server/src/main/java/org/elasticsearch/search/query/QueryPhase.java

```diff
@@ -429,8 +429,7 @@ private static Query tryRewriteLongSort(SearchContext searchContext, IndexReader
         // check if this is a field of type Long or Date, that is indexed and has doc values
         String fieldName = sortField.getField();
         if (fieldName == null) return null; // happens when _score or _doc is the 1st sort field
-        if (searchContext.mapperService() == null) return null; // mapperService can be null in tests
```

Comment:
this made me nervous , but I tried removing, I do expect test failures caused by it, but let's see. Same for other changes in the test classes.

Reply:
:+1: